### DISCUSSION TITLE
How to override environment variables after build / where to define them

### DISCUSSION THREAD

**Discussion by romanzipp** [nuxt/nuxt#20860] [Questions]
I'm having some questions about environment variables availability and overriding during build/runtimes.

My current setup:
- The Nuxt application is built in a CI to a Docker container
- Default SSR - no static site
- The container gets deployed in a high availability Nomad cluster

The Nuxt config would look as followed:

```js
export default defineNuxtConfig({
    runtimeConfig: {
        public : {
            foobar: process.env.FOOBAR
        }
    }
});
```

My current understanding is, that all environment variables must be defined on **build time** - in the CI process.
I realised this by allowing environment variables to be passed as build-args to the container.

```dockerfile
ARG FOOBAR
```

In this particular example, I've set a `FOOBAR` env var to value `a` and built the app. In deployment, I've changed the env var to `b` (confirmed with shell `env` command) but it's still set to `a`.

### Questions

- Can i override the config by setting the `FOOBAR` env var in the container at runtime? (from my current testing - no)
- Would it change anything if I would not provide the env vars at build time?
- Is there any effect if I add or omit the `NUXT_` prefix to env vars at runtime? (as I've understood this only matters if you haven't defined them with `process.env.*` [docs](https://nuxt.com/docs/guide/going-further/runtime-config))
- If any answer to the last questions was yes - would this config change also be reflected in the client assets?

Thanks for you help

**Comment by maou-shonen** [comment]
https://nuxt.com/docs/guide/going-further/runtime-config#environment-variables
`NUXT_` prefix -> Runtime override
Other environment variables -> Do nothing
`process.env` is hard-coded into dist during build.
